### ISSUE TITLE
Excludes for variable-name

### ISSUE THREAD

**Issue by Anahkiasen** [palantir/tslint#522]
I have a variable in my app that comes from a third party and needs to be named a certain way. Would it be possible to add the ability to exclude a certain variable name? Like,

``` js
"variable-name": [true,
    "TheVarName",
],
```

Or something like that?

**Comment by sundaysen** [comment]
You can insert rule exclusions in your code using comments
```
/* tslint:disable: name-of-rule-to-disable */
yourCodeHere
/* tslint:enable */
```

**Comment by Anahkiasen** [comment]
Yes but in my case it's a service I import in multiple controllers and I don't want to litter my codes with comments like that, it would be way easier to ignore it globally. If the syntax is the issue then maybe an object (which would be clearer anyway than the array thing used currently):

```js
"variable-name": {
  enabled: true,
  ignore: ['TheVarName'],
},
```

**Comment by adidahiya** [comment]
What's the exact code this rule is failing on? Are you re-declaring the variable? If it's provided by a third party library, shouldn't it be declared outside of the code you're linting?

**Comment by Anahkiasen** [comment]
It's an Angular service I'm injecting so the variable name has to match the service name registered on the container by the third party

**Comment by adidahiya** [comment]
ah, ok. This is similar to #457. Do you have use cases where the injected variable is used as something _other_ than a constructor? Meaning, if we resolve #457, would that satisfy your use case completely?

**Comment by Anahkiasen** [comment]
Hm I think it might, not sure, it's a factory. Here's my code dumbed down:

```ts
class MyClass {
  constructor(public Flash: Flash) {}

  someMethod() {
    this.Flash.create('success', 'The message');
  }
```

**Comment by adidahiya** [comment]
Even with #577 implemented, that still doesn't address variables from third party libs which are declared as interfaces instead of classes. So I agree we should implement excludes for the variable-name rule.

**Comment by ajcrites** [comment]
Another example of when this comes up is using stateless functional components in react.

    export const Example: React.SFC<{}> = () => <div />;

With the current rule, this variable name doesn't pass the lint check, but React requires that custom components including SFCs be capitalized.

**Comment by ChrisSG** [comment]
The SFC is what hits us at the moment. I think that maybe only classnames should be PascalCase and one should explicitely do it like this:

const example = ...

export {example as Example}

Any more thoughts on this?

**Comment by joshghent** [comment]
We have also run into this issue at my work place, we do not use snake_case variables but have multiple data base columns that are snake_case and therefore violate the rules. The solution outlined in the original post (or something similar) would solve this for us because we know exactly what we want to allow.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
